extensions/project-descriptor.md

```diff
@@ -8,80 +8,120 @@ A project descriptor is a file that MAY contain configuration for apps, services
 - [Project Descriptor](#project-descriptor)
   - [Table of Contents](#table-of-contents)
   - [Schema](#schema)
-  - [`[project]`](#project)
-  - [`[[project.licenses]]`](#projectlicenses)
-  - [`[build.include]` and `[build.exclude]`](#buildinclude-and-buildexclude)
-  - [`[[build.buildpacks]]`](#buildbuildpacks)
-  - [`[[build.env]]`](#buildenv)
-  - [`[metadata]`](#metadata)
+    - [`[_]`](#_)
+      - [`[_]`](#_-1)
+      - [`[[_.licenses]]`](#_licenses)
+      - [`[_.metadata]`](#_metadata)
+    - [Non-`[_]` Tables](#non-_-tables)
+    - [`[io.buildpacks]`](#iobuildpacks)
+      - [`[io.buildpacks.build.include]` and `[io.buildpacks.build.exclude]`](#iobuildpacksbuildinclude-and-iobuildpacksbuildexclude)
+      - [`[[io.buildpacks.build.buildpacks]]`](#iobuildpacksbuildbuildpacks)
+      - [`[[io.buildpacks.build.env]]`](#iobuildpacksbuildenv)
   - [Example](#example)
 
 ## Schema
 
-The TOML schema of the project descriptor is the following:
+### `[_]`
+
+API Version: `0.2`
+
+The TOML schema of the project section of the project descriptor:
 
 ```toml
-[project]
+[_]
+api = "<string>"
 id = "<string>" # machine readable
 name = "<string>" # human readable
 version = "<string>"
 authors = ["<string>"]
 documentation-url = "<url>"
 source-url = "<url>"
 
-[[project.licenses]]
+[[_.licenses]]
 type = "<string>"
 uri = "<uri>"
 
-[build]
-include = ["<string>"]
-exclude = ["<string>"]
-[[build.buildpacks]]
-id = "<string>"
-version = "<string>"
-uri = "<string>"
-[[build.env]]
-name = "<string>"
-value = "<string>"
-[metadata]
+[_.metadata]
 # additional arbitrary keys allowed
 ```
 
 The following sections describe each part of the schema in detail.
 
-## `[project]`
-
-The top-level `[project]` table MAY contain configuration about the repository, including `id` and `version`. It MAY also include metadata about how it is authored, documented, and version controlled.
+#### `[_]`
 
-The `project.id`
+The top-level `_` table MAY contain configuration about the repository, including `id` and `version`. It MAY also include metadata about how it is authored, documented, and version controlled. It MUST contain `api`  to denote which version of this API the descriptor is using.
 
 ```toml
-[project]
+[_]
+api = "<string>"
 id = "<string>"
 name = "<string>"
 version = "<string>"
+authors = ["<string>"]
+documentation-url = "<url>"
+source-url = "<url>"
```

Comment:
I feel like we are a bit all over the place with the values in this schema. Sometime they are a:
1. TOML type (e.g `"<string>"`)
2. a type but not a TOML primitive (e.g `<"url">`)
3. example (e.g `"JAVA_OPTS"`)

And some optional values like`"<buildpack ID (optional)>"` include `(optional)`, but we don't do this for every optional value.

I think we should pick a strategy and be consistent.

Reply:
@ekcasey is there a strategy we're doing for the entire spec? (we probably need spec guidelines)